### PR TITLE
24765 add support for enum props

### PR DESCRIPTION
**Pull request by frowe**
Issue #24765

### CODE REVIEW

File: dev/com.ibm.ws.jdbc.metatype/resources/OSGI-INF/metatype/metatype.xml [hunk 1]
```diff
@@ -1397,6 +1397,11 @@
   <AD id="tcpKeepAlive"                  ibmui:group="Advanced" required="false" type="Boolean" name="%tcpKeepAlive" description="%tcpKeepAlive.desc"/>
   <AD id="user"                          ibmui:group="Advanced" required="false" type="String"  name="%user" description="%user.desc"/>
   <AD id="URL"                           ibmui:group="Advanced" required="false" type="String"  name="%URL" description="%URL.desc"/>
+  <AD id="autosave"                      ibmui:group="Advanced" required="false" type="String"                         name="%autosave"             description="%sslMode.desc">
```
Comment (KyleAure) [inline dev/com.ibm.ws.jdbc.metatype/resources/OSGI-INF/metatype/metatype.xml:1400]:
```suggestion
  <AD id="autosave"                      ibmui:group="Advanced" required="false" type="String"                         name="%autosave"             description="%autosave.desc">
```

File: dev/com.ibm.ws.jdbc/src/com/ibm/ws/jdbc/internal/JDBCDriverService.java [hunk 2]
```diff
@@ -1044,6 +1044,9 @@ else if (paramType.equals(Character.class)) // special case: Character
 
             else if(paramType.equals(char[].class)) // special case: char array
                 param = value.toCharArray();
+            
+            else if (paramType.isEnum()) // special case: Enum
+                param = Enum.valueOf((Class<Enum>)paramType, value.toUpperCase());
```
Comment (KyleAure) [inline dev/com.ibm.ws.jdbc/src/com/ibm/ws/jdbc/internal/JDBCDriverService.java:1049]: Should we add some exception handling here? 
It is possible enum values could be removed in future driver updates which will result in an `IllegalArgumentException` being thrown. We should handle that exception and perhaps add a warning that `this version of the JDBC driver doesn't support value x`

**Review by KyleAure — CHANGES_REQUESTED** [review]

**Comment by posmith** [comment]
L2 message review not required.

**Comment by KyleAure** [comment]
Closing in favor of #24834 
We had some issues with remote forks and need to abandon this PR.

**Pull request closed without merging**